src/scenic/syntax/compiler.py

```diff
@@ -43,3 +52,156 @@ def visit_New(self, node: s.New):
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_DistanceFromOp(self, node: s.DistanceFromOp):
+        return ast.Call(
+            func=ast.Name(id="DistanceFrom", ctx=loadCtx),
+            args=[self.visit(node.target)],
+            keywords=[]
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_DistancePastOp(self, node: s.DistancePastOp):
+        return ast.Call(
+            func=ast.Name(id="DistancePast", ctx=loadCtx),
+            args=[self.visit(node.target)],
+            keywords=[]
+            if node.base is None
+            else [ast.keyword(arg="Y", value=self.visit(node.base))],
+        )
+
+    def visit_AngleFromOp(self, node: s.AngleFromOp):
+        if node.base is None and node.target is None:
+            assert False, "neither target nor base were specified in AngleFromOp"
+        base = ego if node.base is None else self.visit(node.base)
+        target = ego if node.target is None else self.visit(node.target)
+        return ast.Call(
+            func=ast.Name(id="AngleFrom", ctx=loadCtx),
+            args=[base, target],
+            keywords=[],
+        )
+
+    def visit_FollowOp(self, node: s.FollowOp):
+        return ast.Call(
+            func=ast.Name(id="Follow", ctx=loadCtx),
+            args=[
+                self.visit(node.target),
+                self.visit(node.base),
+                self.visit(node.distance),
+            ],
+            keywords=[],
+        )
+
+    def visit_VisibleOp(self, node: s.VisibleOp):
+        return ast.Call(
+            func=ast.Name(id="Visible", ctx=loadCtx),
+            args=[self.visit(node.region)],
+            keywords=[],
+        )
+
+    def visit_NotVisibleOp(self, node: s.VisibleOp):
+        return ast.Call(
+            func=ast.Name(id="NotVisible", ctx=loadCtx),
+            args=[self.visit(node.region)],
+            keywords=[],
+        )
+
+    def visit_PositionOfOp(self, node: s.PositionOfOp):
+        classToFunctionName = {
+            s.Front: "Front",
+            s.Back: "Back",
+            s.Left: "Left",
+            s.Right: "Right",
+            s.FrontLeft: "FrontLeft",
+            s.FrontRight: "FrontRight",
+            s.BackLeft: "BackLeft",
+            s.BackRight: "BackRight",
+        }
+        f = None
+        for c, n in classToFunctionName.items():
+            if isinstance(node.position, c):
+                f = n
+                break
+        assert f is not None, f"impossible position {node.position.__class__.__name__}"
+        return ast.Call(
+            func=ast.Name(id=f, ctx=loadCtx),
+            args=[self.visit(node.target)],
+            keywords=[],
+        )
+
+    def visit_DegOp(self, node: s.DegOp):
+        return ast.BinOp(
+            left=self.visit(node.operand), op=ast.Mult(), right=ast.Constant(0.01745329)
```

Comment:
I wonder if having a magic number here is a good idea. For example, we can replace the magic number with `math.radian(1)`

Or better, we can make it that this visitor replaces the node to a call to `math.radian` with the operand as an argument. Because of how `math.radian` is implemented, this will not change the result at all. https://github.com/python/cpython/blob/605e9c66ad367b54a847f9fc65447a071742f554/Modules/mathmodule.c#L2938-L2943

Reply:
Agreed, it's clearer to call `math.radians`. This could break if someone bound the name `math`, but maybe we should put off fixing that for now as the same problem occurs with other names imported from the veneer: e.g. if someone defined something called `CanSee` it would break the `can see` operator. Probably we should move all such names into attributes of some name which is unlikely to collide, e.g. `__scenic.CanSee`.

Reply:
It turns out I cannot use `math` inside the compiled Python AST because it is not included. If I want to use `math` I need to either

1. Import `math` from `veneer`, write a `Deg` function that wraps `math.radian` and call that function, or
2. Add `import math` (or similar) to [`preamble`](https://github.com/BerkeleyLearnVerify/Scenic/blob/952609262e4f9178b1a5c6956c6ba0de7dbfe776/src/scenic/syntax/translator.py#L310-L315)

both of which I think go beyond the scope of this PR. So I will leave it as is for now and added to the TODO list.

It feels like `from scenic.syntax.veneer import *` is the root cause of the name collision issue you mentioned. We can change it to something like `from scenic.syntax.veneer import __scenic`, and then we can add a `__scenic.` to every function call that the compiler generates. I will have to see if this change has side effects, but looks promising to me. Also added this to the TODOs.

Reply:
OK. Yes, just importing a single name from the veneer and routing everything through that is what I had in mind. It should be OK since we only ever read from the imported names, never rebind them.